feat(native): Add local merge spilling session properties (#26849)
prestodb/presto: presto-docs/src/main/sphinx/presto_cpp/properties-session.rst

Native Execution only. Enable topN row number spilling on native engine.

``native_local_merge_spill_enabled``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

* **Type:** ``boolean``
* **Default value:** ``false``

Native Execution only. Enable local merge spilling on native engine.

``native_local_merge_max_num_merge_sources``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

* **Type:** ``integer``
* **Default value:** ``2147483647``

Native Execution only. Specifies the maximum number of local sources to merge at a time.

> This can be used to limit memory usage during local merge operations.

Reviewer: ```suggestion
Use this property to limit memory usage during local merge operations.
```
Suggesting active voice here.
